### PR TITLE
[CSL-2230] Update quizzes endpoints

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

File: src/types/quizzes.d.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
-import { ConstructorClientOptions, NetworkParameters, UserParameters } from '.';
```
Comment (esezen) [inline src/types/quizzes.d.ts:1]: Not related to this line. Can we update `src/types/tests/quizzes.test-d.ts`? `npm run test:types` is failing ATM

File: src/types/quizzes.d.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+import { ConstructorClientOptions, NetworkParameters, UserParameters, SortOption, ResultSources, Group, Facet, Feature } from '.';
 
 export default Quizzes;
 
 export interface QuizzesParameters {
   section?: string;
   answers?: any[];
```
Comment (esezen) [inline src/types/quizzes.d.ts:7]: Not related to this PR but can we make this required. It's already required in the implementation

**Review by esezen — CHANGES_REQUESTED** [review]
Implementation is looking great. Left some comment about docs and types

File: src/modules/quizzes.js [hunk 3]
```diff
@@ -88,12 +93,12 @@ class Quizzes {
    *
    * @function getQuizNextQuestion
    * @description Retrieve quiz question from Constructor.io API
-   * @param {string} id - The identifier of the quiz
+   * @param {string} quizId - The identifier of the quiz
```
[the remark this answers is not in the document]
Comment (mocca102) [reply src/modules/quizzes.js:96]: it's quizId, quizVersionId, quizSessionId and it's consistent like this

[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/quizzes.js:96]: Yeah that makes sense if it's quizId in other SDKs. @jjl014 wdyt?

[the remark this answers is not in the document]
Comment (jjl014) [reply src/modules/quizzes.js:96]: Agreed with @mocca102, let's make it `quizId` here. I didn't realize our JS client had it as `id`. Created a story [here](https://linear.app/constructor/issue/CSL-2324/js-client-sdk-update-id-to-quizid-for-quizzes-module) for us to work on updating it

**Comment by stanlp1** [comment]
Updates types ( using your updates from -javascript repo ) 

Wonder if we can just create on repo with types for result endpoints. Since we're currently maintaining it in multiple places right now 🤔

File: src/types/quizzes.d.ts [hunk 4]
```diff
@@ -1,11 +1,31 @@
...
+  SortOption,
+  FilterExpression,
+  ResultSources,
+} from '.';
 
 export default Quizzes;
 
 export interface QuizzesParameters {
   section?: string;
   answers?: any[];
```
Comment (esezen) [inline src/types/quizzes.d.ts:19]: Not related to this PR but can we make this required. It's already required in the implementation

File: src/modules/quizzes.js [hunk 5]
```diff
@@ -164,11 +170,12 @@ class Quizzes {
    *
    * @function getQuizResults
    * @description Retrieve quiz recommendation and filter expression from Constructor.io API
-   * @param {string} id - The identifier of the quiz
+   * @param {string} quizId - The identifier of the quiz
    * @param {string} [parameters] - Additional parameters to refine result set
```
Comment (esezen) [inline src/modules/quizzes.js:174]: Since a child of `parameters` is required can we also remove [] around parameters here?

**Review by esezen — APPROVED** [review]
LGTM! 🚀

File: src/modules/quizzes.js [hunk 6]
```diff
@@ -111,15 +117,15 @@ class Quizzes {
    *    versionId: '123'
```
Comment (jjl014) [inline src/modules/quizzes.js:117]: Let's update the example too 👍 `versionId` => `quizVersionId`

File: src/modules/quizzes.js [hunk 7]
```diff
@@ -188,15 +195,15 @@ class Quizzes {
...
```
Comment (jjl014) [inline src/modules/quizzes.js:195]: Let's update the example too 👍  `versionId` => `quizVersionId`

File: src/modules/quizzes.js (hunk 5 above)
Comment (jjl014) [reply src/modules/quizzes.js:174]: Agreed, let's update this

File: src/types/quizzes.d.ts [hunk 8]
```diff
@@ -16,38 +36,89 @@ declare class Quizzes {
   getQuizNextQuestion(
     id: string,
     parameters?: QuizzesParameters,
-    userParameters?: UserParameters,
     networkParameters?: NetworkParameters
   ): Promise<NextQuestionResponse>;
 
   getQuizResults(
     id: string,
```
Comment (jjl014) [inline src/types/quizzes.d.ts:43]: Does this need to be changed to `quizId`? 🤔

File: src/types/quizzes.d.ts [hunk 9]
```diff
@@ -16,38 +36,89 @@ declare class Quizzes {
...
 
 /* quizzes results returned from server */
 export interface NextQuestionResponse extends Record<string, any> {
-  next_question: Partial<NextQuestion>;
+  next_question: Question;
   is_last_question?: boolean;
-  version_id?: string;
+  quiz_version_id?: string;
+  quiz_id?: string;
+  quiz_session_id?: string;
```
Comment (jjl014) [inline src/types/quizzes.d.ts:55]: Does this mean they are optional? I believe they should be returned with every response right? 🤔

File: src/modules/quizzes.js [hunk 10]
```diff
@@ -164,11 +170,12 @@ class Quizzes {
...
    * @function getQuizResults
    * @description Retrieve quiz recommendation and filter expression from Constructor.io API
-   * @param {string} id - The identifier of the quiz
+   * @param {string} quizId - The identifier of the quiz
    * @param {string} [parameters] - Additional parameters to refine result set
+   * @param {array} parameters.answers - An array of answers in the format [[1,2],[1]]
    * @param {string} [parameters.section] - Product catalog section
-   * @param {array} [parameters.answers] - An array of answers in the format [[1,2],[1]]
-   * @param {string} [parameters.versionId] - Specific version identifier for the quiz
+   * @param {string} [parameters.quizVersionId] - Version identifier for the quiz. Version ID will be returned with the first request and it should be passed with subsequent requests. More information can be found: https://docs.constructor.io/rest_api/quiz/using_quizzes/#quiz-versioning
```
Comment (jjl014) [inline src/modules/quizzes.js:177]: For the places where we link to docs, could we use the format `[link text]{@link namepathOrURL}`

i.e. `[here]{@link https://docs.constructor.io/rest_api/quiz/using_quizzes/#quiz-versioning}`

File: src/types/quizzes.d.ts [hunk 11]
```diff
@@ -16,38 +36,89 @@ declare class Quizzes {
   getQuizNextQuestion(
     id: string,
```
Comment (jjl014) [inline src/types/quizzes.d.ts:37]: Does this need to be changed to `quizId`? 🤔

**Review by jjl014 — COMMENTED** [review]
Nice, looking pretty sweet to me. Just left a couple comments.

I'm not super familiar with types, so I'm going to lean on you guys to sus things out here. The tests seems to be failing as well when I run `npm run test:types`. Might need to look into that.

File: src/modules/quizzes.js (hunk 6 above)
Comment (esezen) [reply src/modules/quizzes.js:117]: I remember specifically searching for `versionId` and somehow missed this 🤦

**Review by jjl014 — APPROVED** [review]
Looking good to me! 🚀

**Pull request merged**